lib/weka/core/instances.rb

```diff
@@ -241,6 +262,24 @@ def instance_from(instance_or_values, weight:)
           DenseInstance.new(data, weight: weight)
         end
       end
+
+      def map_attribute_type(type)
+        type = type.downcase.to_sym
+        return nil unless Attribute::TYPES.include? type
+
+        case type
+        when :numeric
+          Attribute::NUMERIC
+        when :nominal
+          Attribute::NOMINAL
+        when :string
+          Attribute::STRING
+        when :date
+          Attribute::DATE
+        else
+          # error, ring the bell?
+        end
+      end
```

Comment:
With using a little bit of self-refelction you could write:

```ruby
Attribute.const_get(type.upcase)
```

which replaces the whole case statement. An else handling isn't needed, because this is already covered with the `unless include?(...)` above.

(Maybe type shouldn't be overwritten, so that you don't have to cast it back into a string before upcasing...)

so it would boil down to sth. like:

```ruby
def map_attribute_type(type)
  return unless Attribute::TYPES.include?(type.downcase.to_sym)
  Attribute.const_get(type.upcase)
end
```